src/routes/activityReports/handlers.js

```diff
@@ -458,7 +458,9 @@ export async function saveReport(req, res) {
 
     newReport.lastUpdatedById = userId;
 
-    const savedReport = await createOrUpdate(newReport, report);
+    // join the updated report with the model object retrieved from the API
+    // since we may not get all fields in the request body
+    const savedReport = await createOrUpdate({ ...report, ...newReport }, report);
```

Comment:
Can you add a test that verifies partial updates don't overwrite fields that aren't included in the update?